src/core/client/admin/components/StoryInfoDrawer/RecacheStoryAction.tsx

```diff
@@ -0,0 +1,46 @@
+import { Localized } from "@fluent/react/compat";
+import RecacheStoryMutation from "coral-admin/components/StoryInfoDrawer/RecacheStoryMutation";
+import { useMutation } from "coral-framework/lib/relay";
+import { Button, Flex } from "coral-ui/components/v2";
+import React, { FunctionComponent, useCallback, useState } from "react";
+
+export interface Props {
+  storyID: string;
+}
+
+const CacheStoryAction: FunctionComponent<Props> = ({ storyID }) => {
+  const [triggered, setTriggered] = useState(false);
+  const recacheStory = useMutation(RecacheStoryMutation);
+  const onRecache = useCallback(async () => {
+    if (triggered) {
+      return;
+    }
+
+    await recacheStory({ id: storyID });
+    setTriggered(true);
```

Comment:
Does this ever get set back to `false` when recaching completes? In my testing, it seems to continue to say `Recaching` until the story info drawer is closed and reopened.

Reply:
Made it so this can refetch the cache state, show that it's cached, then after a time, switch back so user can "recache again" if they feel like it. Similarly for the invalidate cache action.